WordPress/Classes/Networking/ReaderTopicServiceRemote.m

```diff
@@ -174,9 +168,24 @@ - (NSString *)sanitizeTopicNameForAPI:(NSString *)topicName
         topicName = [topicName stringByReplacingOccurrencesOfString:@"--" withString:@"-"];
     }
 
+    topicName = [topicName stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+
     return topicName;
 }
 
+- (NSArray *)normalizeMenuTopicsList:(NSArray *)rawTopics subscribed:(BOOL)subscribed recommended:(BOOL)recommended
+{
+    NSMutableArray *topics = [NSMutableArray array];
+    for (NSDictionary *topicDict in rawTopics) {
+        // Failsafe
+        if (![topicDict isKindOfClass:[NSDictionary class]]) {
+            continue;
+        }
+        [topics addObject:[self normalizeMenuTopicDictionary:topicDict subscribed:subscribed recommended:recommended]];
+    }
+    return [topics copy]; // Return immutable array.
```

Comment:
I am not absolutely sure how this works internally, but if we use `copy` here, wouldn't it create copies for the objects inside the array as well? If so, and if this is what we want, should we implement `NSCopying` protocol for the `RemoteReaderTopic`. If we don't want to actually copy them, can we just return the `topics` directly, since we can't mutate it without casting to `NSMutableArray` by default anyway.